doc/source/developing/testing.rst

```diff
@@ -3,503 +3,400 @@
 
-We attempt to make yt compatible with a wide variety of upstream software
-versions. However, sometimes a specific version of a project that yt depends on
-causes some breakage and must be blacklisted in the tests or a more
-experimental project that yt depends on optionally might change sufficiently
-that the yt community decides not to support an old version of that project.
+.. code-block:: bash
 
-To handle cases like this, the versions of upstream software projects installed
-on the machines running the yt test suite are pinned to specific version
-numbers that must be updated manually. This prevents breaking the yt tests when
-a new version of an upstream dependency is released and allows us to manage
-updates in upstream projects at our pace.
+    $ pytest yt/visualization/tests/test_plotwindow.py::test_all_fields
 
-If you would like to add a new dependency for yt (even an optional dependency)
-or would like to update a version of a yt dependency, you must edit the
-``tests/test_requirements.txt`` file, this path is relative to the root of the
-repository. This file contains an enumerated list of direct dependencies and
-pinned version numbers. For new dependencies, simply append the name of the new
-dependency to the end of the file, along with a pin to the latest version
-number of the package. To update a package's version, simply update the version
-number in the entry for that package.
+See this `link <https://docs.pytest.org/en/stable/usage.html?highlight=invocation>`_ for more on how to invoke pytest.
 
-Finally, we also run a set of tests with "minimal" dependencies installed. Please make sure any new tests you add that depend on an optional dependency are properly set up so that the test is not run if the dependency is not installed. If for some reason you need to update the listing of packages that are installed for the "minimal" dependency tests, you will need to edit ``tests/test_minimal_requirements.txt``.
 
-How to Write Unit Tests
-^^^^^^^^^^^^^^^^^^^^^^^
+Unit Test Tools
+^^^^^^^^^^^^^^^
 
-yt provides several pieces of testing assistance, all in the ``yt.testing``
-module.  Describing them in detail is somewhat outside the scope of this
-document, as in some cases they belong to other packages.  However, a few come
+yt provides several tools to assist with writing unit tests. These tools all reside in the ``yt.testing``
+module.  Describing them all in detail is somewhat outside the scope of this
+document. However, a few come
 in handy:
 
 * :func:`~yt.testing.fake_random_ds` provides the ability to create a random
   dataset, with several fields and divided into several different
-  grids, that can be operated on.
+  grids.
 * :func:`~yt.testing.assert_equal` can operate on arrays.
 * :func:`~yt.testing.assert_almost_equal` can operate on arrays and accepts a
   relative allowable difference.
 * :func:`~yt.testing.assert_allclose_units` raises an error if two arrays are
```

Comment:
I agree it's simpler to just refer to the module, but I'd still like to have numpy mentioned, since they do the heavy lifting. Btw I'm not sure those qualify as "wrappers", it looks to me like they are merely aliased (and I still don't see the reason _why_ it is so, but that's out of scope here)